lib/minizinc_data.ex

```diff
@@ -121,11 +122,21 @@ defmodule MinizincData do
   defp array_to_dzn(el, bases) do
     dims = dimensions(el)
 
-    if dims do
-      array_dimensions(dims, make_base_list(dims, bases)) <>
-        "[#{Enum.join(List.flatten(el), @element_separator)}]" <> ")"
-    else
-      throw({:irregular_array, el})
+    case dims do
+      [_] ->
+        # 1D
+        "[#{array_row_to_dzn(el)}]"
+
+      [_, _] ->
+        # 2D
+        "[|#{Enum.map_join(el, @row_separator, &array_row_to_dzn/1)}|]"
+
+      [_ | _] ->
+        array_dimensions(dims, make_base_list(dims, bases)) <>
+          "[#{Enum.join(List.flatten(el), @element_separator)}]" <> ")"
```

Comment:
My reason for switching to the other format here was actually not directly related to the sets.  Here's a small sample MZN:
```
enum keys;
array[keys] of bool: truth_values;

constraint true
solve satisfy;
```

and with Elixir data:
```elixir
%{
  keys: {:a, :b, :c},
  truth_values: [true, false, true]
}
```

The earlier generated DZN looked like `array1d(1..3, [true, false, true])` which MiniZinc rejected. `[| true, false, true|]` is properly coerced to the enum keys.

Reply:
Thanks, I see the issue. It should have generated:
```
truth_values = array1d(keys, [true, false, true]);
```

Reply:
That would have worked as well, but it appears that all dimensions are converted to ranges?

Reply:
Correct. Yep, enums need a little bit of work. Dimensions are currently inferred by the shape of the lists. For arrays with enum ranges, I'm thinking we could pass the enum par names instead of bases.